packages/cloud-cognitive/src/components/ButtonSetWithOverflow/ButtonSetWithOverflow.js

```diff
@@ -0,0 +1,216 @@
+    checkFullyVisibleItems();
+  };
+
+  const handleButtonResize = () => {
+    /* istanbul ignore next */ // not sure how to test resize
+    checkFullyVisibleItems();
+  };
+
+  useEffect(() => {
+    if (buttons) {
+      setItemArray(buttons);
+    } else {
+      const unwrapped = unwrapIfFragment(children);
+      setItemArray(
+        unwrapped.map((item) => {
+          const { children: label, ...rest } = item.props;
+          return { label, ...rest };
+        })
+      );
+    }
+  }, [buttons, children]);
+
+  const AButtonSet = React.forwardRef(({ buttons, size, ...rest }, ref) => {
+    return (
+      <ButtonSet {...rest} ref={ref}>
+        {buttons.map(({ label, ...other }, index) => (
+          <Button key={index} {...other} size={size} type="button">
+            {label}
+          </Button>
+        ))}
+      </ButtonSet>
+    );
+  });
+  const ATempComboButton = React.forwardRef(
+    ({ buttons, size, ...rest }, ref) => {
+      return (
+        <TempComboButton
+          {...rest}
+          buttons={buttons}
+          size={size}
+          ref={ref}
+          label={pageActionsLabel}
+        />
+      );
+    }
+  );
+
+  return (
+    <ReactResizeDetector handleWidth={true} onResize={handleResize}>
+      <div className={cx([blockClass, className])} ref={spaceAvailableRef}>
+        <ReactResizeDetector onResize={handleButtonResize}>
+          <div
+            className={`${blockClass}__button-container ${blockClass}__button-container--hidden`}>
+            <AButtonSet
+              aria-hidden={true}
+              ref={sizingContainerRefSet}
+              size={size}
+              buttons={itemArray}
+            />
+          </div>
+        </ReactResizeDetector>
+        <ReactResizeDetector onResize={handleButtonResize}>
+          <div
+            ref={sizingContainerRefCombo}
+            className={`${blockClass}__button-container ${blockClass}__button-container--hidden`}
+            aria-hidden={true}>
+            <ATempComboButton buttons={itemArray} size={size} />
+          </div>
+        </ReactResizeDetector>
+
+        {showAsOverflow ? (
+          <ATempComboButton buttons={itemArray} size={size} />
+        ) : (
+          <AButtonSet
+            className={`${blockClass}__button-container`}
+            size={size}
+            buttons={itemArray}
+          />
+        )}
+      </div>
+    </ReactResizeDetector>
+  );
+};
+
+ButtonSetWithOverflow.propTypes = {
+  /**
+   * Button shape things for us to render - see Carbon button kind, label, onClick.
+   */
+  buttons: PropTypes.oneOfType([
+    PropTypes.arrayOf(
+      PropTypes.shape({
+        kind: PropTypes.oneOf([
+          'primary',
+          'secondary',
+          'danger',
+          'ghost',
+          'danger--primary',
+          'danger--ghost',
+          'danger--tertiary',
+          'tertiary',
+        ]),
+        label: PropTypes.node,
+        onClick: PropTypes.func,
+      })
+    ),
+  ]),
```

Comment:
```suggestion
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      kind: PropTypes.oneOf([
        'primary',
        'secondary',
        'danger',
        'ghost',
        'danger--primary',
        'danger--ghost',
        'danger--tertiary',
        'tertiary',
      ]),
      label: PropTypes.node,
      onClick: PropTypes.func,
    })
  ),
```

The oneOfType is only needed to include a validator -- you could leave it in if you're intended to add a validator later.